src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let final = '';
+    const joinSeparator = separator;
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      if (element === null || element === undefined) {
+        element = ''; // code doesn't pass tests if I skip this iteration
```

Comment:
Add comments to PR, not in the code directly 
```suggestion
        element = '';
```